MomsSpaghetti/Pages/Index.cshtml.cs

```diff
@@ -257,48 +259,65 @@ public ActionResult OnPostDeleteLiked()
         }
 
 
-    
-        public async Task<IActionResult>  OnPost()
-        {   
-            String responseContent="";
+
+        public async Task<IActionResult> OnPost()
+        {
+            String responseContent = "";
             var responseObject = new { };
+            if (SearchRecipes.Query == null)
+            {
+
+                return Page();
+            }
             if (SearchRecipes.Query.Length != 0)
             {
-                
+
                 try
                 {
                     HttpClient client = new HttpClient();
 
                     // client.DefaultRequestHeaders.Add("q", UserInput);
 
-                    HttpResponseMessage response = await client.GetAsync("https://forkify-api.herokuapp.com/api/search?q="+ SearchRecipes.Query);
+                    HttpResponseMessage response = await client.GetAsync("https://forkify-api.herokuapp.com/api/search?q=" + SearchRecipes.Query);
 
                     if (response.IsSuccessStatusCode)
                     {
-                       responseContent = await response.Content.ReadAsStringAsync();
+                        responseContent = await response.Content.ReadAsStringAsync();
                         JObject jsonObject = JObject.Parse(responseContent);
-                        //TempData["searchResultsString"] = responseContent;
-                        //UserSearch SearchRecipes = new UserSearch();
-                        SearchRecipes.Result = jsonObject;
-                        // ViewData["result"] = SearchRecipes.Result["recipes"][0]["title"];
-                       /* var cacheEntry = _cache.CreateEntry("searchRecipeResults");
-                        cacheEntry.SetValue(SearchRecipes.Result);
-                        cacheEntry.Dispose(); //adds item to cache
-                       */
 
-                        _cache.Set<JObject>("searchRecipeResults", SearchRecipes.Result);
-                        /*  var props = SearchRecipes.Result.Properties();
-                         var  propsArray = props.ToArray();
-                          var first = propsArray[1].Value[0]["title"];  //Upto VALUE[0] gives the first recipe object
+                        //add validation
+                        JSchema schema = JSchema.Parse(System.IO.File.ReadAllText("searchResults.json"));
```

Comment:
Added searchResults.json for validation .
Source: https://medium.com/@schneidenbach/restful-api-best-practices-and-common-pitfalls-7a83ba3763b5